src/celeritas/ext/GeantSetup.cc

```diff
@@ -67,6 +68,22 @@ class DetectorConstruction : public G4VUserDetectorConstruction
 //---------------------------------------------------------------------------//
 }  // namespace
 
+//---------------------------------------------------------------------------//
+/*!
+ * G4RunManager::GetNumberOfThreads isn't virtual before Geant4 v10.7.0 so we
+ * need to explicitely try dynamic cast to G4MTRunManager to get the number of
+ * threads. If the dynamic cast fails, return 1
+ */
+int GetNumberOfThreads(G4RunManager const& runman)
```

Comment:
Celeritas itself has different variable name conventions compared to Geant4 (whose conventions we try to use in `accel`): please rename to `get_num_threads`

Reply:
Since this function should only be used with geant4 version < 10.7.0, should we add `#if` guards to only compile it with old geant4 releases?

Reply:
The function should be used as a version-independent way to access the number of threads from the run manager.

Reply:
Sure, it works for any version. It's just that in 10.7.0 this function isn't needed since `G4RunManager::GetNumberOfThreads` has been declared as virtual instead of only being declared in `G4MTRunManager`

Reply:
OK, I see what the confusion is, sorry. For expedience I've pushed a changeset with what I mean. The idea is just to encapsulate the geant4 version dependency.